rfc/rfc-0001-cli-support-for-publishing-and-subscribing-to-streams.md

```diff
@@ -0,0 +1,51 @@
+# RFC-0001: CLI support for publishing and subscribing to streams
+
+**Authors:** Swapnil Bawaskar
+
+**Status:**
+
+**Pull Request URL:** https://github.com/projectriff/riff/pull/1359
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+When the streaming runtime is installed, riff users can create streams (backed by a stream provider e.g. kafka). Users can write functions that read from and write to these streams,
+but there is no easy mechanism to get events into the stream and look into the contents of the stream which are helpful while developing/demoing.
+
+### Anti-Goals
+We will only address this problem for development/demos, not production, so topics like auth/authz are out of scope for this document.
+
+## Solution
+The idea is to provide users with only one cli (riff cli) that will enable users to send/receive messages to/from a stream (commands below). We will have to ask the users to port-forward the gateway service if they are running outside the cluster.
+
+#### User Impact
+We should introduce the following commands to the riff cli:
+- send a message to the stream:
+
+    `riff streaming stream publish <stream-name> --payload <payload-as-string> --headers <headers-as-string> --payload-file <relative-filepath>`  
+    content type has already been specified while creating the stream.
+    Options `payload` and `payload-file` are mutually exclusive. The former is to be used for strings and the later for binary payloads. The binary payload will be base64 encoded.
+
+- subscribe for messages from a stream:
+
+    `riff streaming stream subscribe <stream-name> --offset <long-offset>`  
+    this will display all the message from the given offset and then block to display further messages, essentially preventing user from entering subsequent commands.
+    The messages will be displayed as json with the following fields:
+    ``` 
+    {
+        "payload": "the users payload",
```

Comment:
Need to clarify how (and why) we would output "complicated payloads" (eg base64 encoding of them)